### PR TITLE
Exclude CAPI deps and suggest apiextensions v4

### PR DESCRIPTION
**Pull request by fiunchinho**
We can't just upgrade the CAPI dependencies, we want to control when this happens. So let's exclude those dependencies from renovate.

Also we want to exclude one faulty apiextensions release 3.40.0, so let's suggest v4.0.0 onwards.

### Checklist

- [X] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Changed
+
+- Renovate excludes cluster-api dependencies.
```
Comment (rossf7) [inline CHANGELOG.md:12]:
```suggestion
- Renovate exclude cluster-api dependencies.
```

Nit but just for consistency with apiextensions entry.

**Review by rossf7 — APPROVED** [review]
LGTM

**Comment by marians** [comment]
I also had https://github.com/giantswarm/devctl/pull/335 in the making. Anyway, thanks!

**Review by marians — APPROVED** [review]

**Comment by fiunchinho** [comment]
> I also had #335 in the making. Anyway, thanks!

Sorry @marians I did not realize that

**Review by MarcelMue — APPROVED** [review]

**Pull request merged**